Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -12,118 +12,97 @@
 #include <AzCore/Component/ComponentApplicationBus.h>
 #include <AzCore/Serialization/EditContext.h>
 
-#include <AzFramework/StringFunc/StringFunc.h>
-
+#include <AzFramework/Archive/INestedArchive.h>
+#include <AzFramework/Archive/ZipDirStructures.h>
 #include <AzFramework/Process/ProcessCommunicator.h>
 #include <AzFramework/Process/ProcessWatcher.h>
 #include <AzFramework/FileFunc/FileFunc.h>
 
+
 namespace AzToolsFramework
 {
-    // Forward declare platform specific functions
-    namespace Platform
-    {
-        AZStd::string GetZipExePath();
-        AZStd::string GetUnzipExePath();
-
-        AZStd::string GetCreateArchiveCommand(const AZStd::string& archivePath, const AZStd::string& dirToArchive);
-        AZStd::string GetExtractArchiveCommand(const AZStd::string& archivePath, const AZStd::string& destinationPath, bool includeRoot);
-        AZStd::string GetAddFileToArchiveCommand(const AZStd::string& archivePath, const AZStd::string& file);
-        AZStd::string GetAddFilesToArchiveCommand(const AZStd::string& archivePath, const AZStd::string& listFilePath);
-        AZStd::string GetExtractFileCommand(const AZStd::string& archivePath, const AZStd::string& fileInArchive, const AZStd::string& destinationPath, bool overWrite);
-        AZStd::string GetListFilesInArchiveCommand(const AZStd::string& archivePath);
-        void ParseConsoleOutputFromListFilesInArchive(const AZStd::string& consoleOutput, AZStd::vector<AZStd::string>& fileEntries);
-    }
+    constexpr const char s_traceName[] = "ArchiveComponent";
+    constexpr AZ::u32 s_compressionMethod = AZ::IO::INestedArchive::METHOD_DEFLATE;
+    constexpr AZ::s32 s_compressionLevel = AZ::IO::INestedArchive::LEVEL_NORMAL;
+    constexpr CompressionCodec::Codec s_compressionCodec = CompressionCodec::Codec::ZLIB;
 
-    const char s_traceName[] = "ArchiveComponent";
-    const unsigned int g_sleepDuration = 1;
-
-    // Echoes all results of stdout and stderr to console and never blocks
-    class ConsoleEchoCommunicator
+    namespace ArchiveUtils
     {
-    public:
-        ConsoleEchoCommunicator(AzFramework::ProcessCommunicator* communicator)
-            : m_communicator(communicator)
+        // Read a file's contents into a provided buffer.
+        // Does not add a zero byte at the end of the buffer.
+        // returns true if read was successful, false otherwise.
+        bool ReadFile(const AZStd::string& filePath, AZ::IO::OpenMode openMode, AZStd::vector<char>& outBuffer)
```

Comment:
Updating the signature of this function to accept a `const AZ::IO::Path& filePath` is more appropriate.